src/invariant/test/ConservationOfLumensTests.cpp

```diff
@@ -202,6 +217,17 @@ TEST_CASE("Account balances unchanged without inflation",
         }
 
         auto keepEnd = entries2.begin() + N / 2;
+
+        // We're going to increase LedgeHeader::totalCoins by the total balance
+        // of the entries starting at `keepEnd` when we call `updateBalances()`
+        // below, so check whether that would overflow, and if so, bump
+        // `keepEnd` until it wouldn't overflow.
+        while (getTotalBalance(std::vector<LedgerEntry>(
```

Comment:
This is a "garbage in, garbage out" situation. The problem is that we feed bad data to the tests in the first place.

Code like `std::generate_n(std::back_inserter(entries1), N, std::bind(generateRandomAccount, 2))` is not good enough in this context:
we have a lot of places where we assume that the sum of all native assets is less than `INT64_MAX`, as a consequence it's just better to generate data that at least meets that constraint.

Maybe pick balances sequentially over the "max left" (starting with `1..INT64_max`)?

Reply:
> Code like std::generate_n(std::back_inserter(entries1), N, std::bind(generateRandomAccount, 2)) is not good enough in this context:
we have a lot of places where we assume that the sum of all native assets is less than INT64_MAX, as a consequence it's just better to generate data that at least meets that constraint.

I think the accounts do meet that constraint -- `generateRandomAccount()`, despite the name, always sets the account balance to `0`, and `updateBalances()` checks `totalCoins` to limit the amount of asset that it can distribute to the accounts in total.

I think Jon nailed the root cause below:

> The root cause of this bug is that we are misusing getTotalBalance.
>
> What the test is trying to do is delete approximately half the entries, transferring the balance randomly to other entries that are not being deleted. This doesn't actually change totalCoins. But getTotalBalance does change totalCoins.

Reply:
(Also see below for the fix.)